django_project/feti/static/js/scripts/models/occupation.js

```diff
@@ -14,10 +16,94 @@ define([
             }
             return data;
         },
+        show: function () {
+            this.renderMarker();
+        },
+        hide: function () {
+            this.removeMarker();
+        },
         destroy: function () {
+            // destroy by remove layers and delete this object
+            this.removeMarker();
             delete this;
         },
-        renderMarker: function () {
+        removeMarker: function () {
+            if (this.get('layer')) {
+                this.get('layer').clearLayers();
+            }
+        },
+        renderMarker: function (data) {
+            this.layerGroup = L.featureGroup();
+            if (!this.get('layer')) {
+                for(var i=0; i<data.length; i++){
+                    var object=data[i];
+                    var location = object['location'];
+                    var public_institution = object['public_institution'];
+                    var markercolor;
+                    if (public_institution) {
+                        markercolor = 'blue leaflet-clickable';
+                    } else {
+                        markercolor = 'red leaflet-clickable';
+                    }
+
+                    var marker = new L.marker([location.coordinates[1], location.coordinates[0]], {
+                        icon: L.ExtraMarkers.icon({
+                            markerColor: markercolor,
+                            icon: 'true',
+                            extraClasses: 'fa fa-graduation-cap',
+                            iconColor: 'white'
+                        })
+                    });
+
+                    var popup = '';
+                    var campus = 'Campus : ' + object['campus'];
+                    var phone = object['phone'];
+
+                    popup = '<div class="leaflet-header"><h3>' +
```

Comment:
could you move this to use html template?

Reply:
done @meomancer